check-pkg/action.yaml

```diff
@@ -26,32 +26,40 @@ runs:
     - name: Check CRAN checks
       id: cran-check-status
       run: |
-        pkg <- "${{ inputs.pkg }}"
-        fp <- paste0("https://flrsh-dev.github.io/cran-checks/", pkg, ".json")
-
-        json <- tryCatch(readLines(fp),
-          warning = function(e) {
-            stop("Package ", pkg, " not found in CRAN checks", call. = FALSE)
-          }
-        )
-
-        res <- yyjsonr::read_json_str(json)
+        pkgs <- unlist(strsplit("${{ inputs.pkg }}", "\\s+"))
+        pkgs <- pkgs[pkgs != ""]
         
-        if (any("ERROR" %in% res$results[[1]]$check_status)) {
-          stop(
-            "Errors found in CRAN checks",
-            "\n",
-            "Review logs at ", 
-            "https://cran.r-project.org/web/checks/check_results_", pkg, ".html"
-          )
-        } else if (any("WARN" %in% res$results[[1]]$check_status)) {
-          stop(
-            "Warnings found in CRAN checks",
-            "\n",
-            "Review logs at ", 
-            "https://cran.r-project.org/web/checks/check_results_", pkg, ".html"
+        pkg_errs <- character(0)
+        pkg_warn <- character(0)
+        for(pkg in pkgs) {
+          fp <- paste0("https://flrsh-dev.github.io/cran-checks/", pkg, ".json")
+        
+          json <- tryCatch(readLines(fp),
+                           warning = function(e) {
+                             stop("Package ", pkg, " not found in CRAN checks", call. = FALSE)
+                           }
```

Comment:
This will stop the for loop in its tracks. I wonder if this should be a warning instead similar to the approach you've taken with `pkg_errs` and `pkg_warn`

Reply:
Ah! Good spot.  I'll make a tweak ...

Reply:
Ok, in commit 489ac38 I've made a tweak, so now my test action:

```{yaml}
name: Check CRAN status

on:
  schedule:
    # Runs daily at 4:00 PM UTC (9:00 AM PST)
    - cron: '0 16 * * *'
  # allows for manually running of the check
  workflow_dispatch:

jobs:
  check_cran_status:
    runs-on: ubuntu-latest

    steps:
      - name: Get CRAN checks (louisaslett fork)
        uses: louisaslett/cran-checks/check-pkg@main
        with:
          pkg: |
            PhaseType
            abn
            mlmc
            data.table
            oiqjwd
            zoltr
```

produces the following output:

```
Error: Cannot find {oiqjwd} on CRAN. Errors found in CRAN checks for {abn}, {data.table}. Warnings found in CRAN checks for {zoltr}.

Review logs at:
  https://cran.r-project.org/web/checks/check_results_abn.html
  https://cran.r-project.org/web/checks/check_results_data.table.html
  https://cran.r-project.org/web/checks/check_results_zoltr.html
```

Do you think that looks ok?